packages/react/src/themes/teams/components/Label/labelVariables.ts

```diff
@@ -1,9 +1,25 @@
 import { pxToRem } from '../../../../lib'
-import { SiteVariablesPrepared, ColorSchemeMapping } from '../../../types'
-import { extendColorScheme } from '../../../colorUtils'
+import {
+  SiteVariablesPrepared,
+  StrictColorSchemeMapping,
+  StrictColorScheme,
+  ComponentAreaName,
+} from '../../../types'
+import { extendColorScheme, pickValuesFromColorScheme } from '../../../colorUtils'
+import { TeamsColorNames } from '../../colors'
+
+// TODO extract this to a util
+const tuple = <T extends ComponentAreaName>(...args: T[]) => args
+const labelColorComponentAreas = tuple('background', 'foreground')
+type LabelColorComponentAreas = typeof labelColorComponentAreas[number]
+
+type LabelColorSchemeMapping = StrictColorSchemeMapping<
+  StrictColorScheme<LabelColorComponentAreas>,
+  TeamsColorNames
+>
```

Comment:
I want us to decide if this is the right approach for defining these, and I will extract this to a separate utility and reuse it in other component variables files.

Reply:
are we expecting this `TeamsColorNames` set to change based on the component type? I think, we might start from exporting simpler `StrictTeamsColorSchemeMapping<TComponentAreas>` type that would do this scaffolding in the background. This will make type interface being easier to client.

Also, we might want to rename this type to something like `TeamsReducedSchemeMapping<TComponentAreas>`

Reply:
I would expect that those will be different across different component, as the text have much more color values then the menu for example. (The menu supports only default and brand colors and we want to use that as typings in the styles.)